Conflicting details on async components with vue router (#1869)
vuejs/docs: src/guide/best-practices/performance.md

const Foo = defineAsyncComponent(() => import('./Foo.vue'))
```

> If using client-side routing via Vue Router, it is strongly recommended to **not** use async components as route components. Async components can still be used inside route components but route component themselves are just dynamic imports. See [Lazy Loading Routes](https://router.vuejs.org/guide/advanced/lazy-loading.html) for more details.

Reviewer: I agree with @skirtles-code. Let's not point to the boilerplate repo approach. Instead encourage using lazy loading by using dynamic imports and then remind the readers that this is slightly different from async components while pointing them to the router docs.

Author: Dynamic imports to handle code splitting and lazy loading, out of the box, makes sense.

The guidance to "encourage the use of asynchronous loading with route components" still seems to unclear on how devs should best achieve this. Are left wondering, is it possible to asynchronously load a router component or is it just the sub-components?

Perhaps this section of the docs could do with separate headings for `Code Splitting & Lazy Loading` and `Loading components asynchronously`. To highlight the separation of the two and how to approach them (for components via `defineAsyncComponent` and router components).

Rather than linking to any external code, was hoping for any short snippet(s) to include as simple examples, or just bullet points of suggested options (Suspense - experimental, limit async to sub-components, any custom loading approach?). To help highlight Async loading is still good but `defineAsyncComponent` should not be used for routes components.

Alternatively, could look at vue router's docs having a section on `Async loading with routes` if it requires more detail, which this page could then link to.

Author: ```suggestion
:::tip
If using client-side routing via [Vue Router](https://router.vuejs.org/), asynchronous loading with route components is strongly recommended, but not with `defineAsyncComponent`. See [Lazy Loading Routes](https://router.vuejs.org/guide/advanced/lazy-loading.html) for more details.
:::
```

Would this be more correct? Then as a separate task can look to add docs in vue router on the options for asynchronous loading with router components.
